File: WarehouseApplication.Server/Models/ProductList.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WarehouseApplication.Server.Models
{
	public class ProductList
	{
		[Key]
		public int ListID{ get; set; }

		[ForeignKey("DeliveryDocument")]
		public int DocumentID { get; set; }
		public DeliveryDocument? Document { get; set; }

		[ForeignKey("Product")]
		public int ProductID { get; set; }
		public Product? Product { get; set; }
		
		public int Quantity { get; set; }
		public decimal Price { get; set; }
	}
}

File: warehouseapplication.client/src/app/models/product-list.ts
import { DeliveryDocument } from './delivery-document';
import { Product } from './product';

export interface ProductList {
  listID?: number;
  documentID?: number;
  document?: DeliveryDocument;
  productID: number;
  product?: Product;
  productName: string;
  productBarcode: string;
  quantity: number;
  price: number;
  isEdit?: boolean;
}
